gqlalchemy/disk_storage.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) 2016-2022 Memgraph Ltd. [https://memgraph.com]
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sqlite3
+import contextlib
+
+from abc import ABC
+from typing import Optional, List
+
+
+class OnDiskPropertyDatabase(ABC):
+    def save_node_property(self, node_id: int, property_name: str, property_value: str) -> None:
+        pass
+
+    def load_node_property(self, node_id: int, property_name: str, property_value: str) -> Optional[str]:
+        pass
+
+    def delete_node_property(self, node_id: int, property_name: str, property_value: str) -> None:
+        pass
+
+    def save_relationship_property(self, relationship_id: int, property_name: str, property_value: str) -> None:
+        pass
+
+    def load_relationship_property(
+        self, relationship_id: int, property_name: str, property_value: str
+    ) -> Optional[str]:
+        pass
+
+    def delete_relationship_property(self, node_id: int, property_name: str, property_value: str) -> None:
+        pass
+
+    def drop_database(self) -> None:
+        pass
+
+
+class SQLitePropertyDatabase(OnDiskPropertyDatabase):
+    def __init__(self, database_path: str):
+        self.database_name = database_path
+        self.create_node_property_table()
+        self.create_relationship_property_table()
+
+    def execute_query(self, query: str) -> List[str]:
+        with contextlib.closing(sqlite3.connect(self.database_name)) as conn:
+            with conn:  # autocommit changes
+                with contextlib.closing(conn.cursor()) as cursor:
+                    cursor.execute(query)
+                    return cursor.fetchall()
+
+    def create_node_property_table(self) -> None:
```

Comment:
I assume they were meant to be private since they are being called in `__init__`
```suggestion
    def _execute_query(self, query: str) -> List[str]:
        with contextlib.closing(sqlite3.connect(self.database_name)) as conn:
            with conn:  # autocommit changes
                with contextlib.closing(conn.cursor()) as cursor:
                    cursor.execute(query)
                    return cursor.fetchall()

    def _create_node_property_table(self) -> None:
```

Reply:
Ammm, create_node_property_table can be private, sure, but there's no harm in exposing it? Execute query should definitely be exposed.